internal/state/current_round_state.go

```diff
@@ -63,8 +63,11 @@ func NewCurrentRoundState(proTxHash types.ProTxHash, rp RoundParams, baseState S
 	return candidate, nil
 }
 
-func (candidate *CurrentRoundState) IsEmpty() bool {
-	return candidate.AppHash == nil
+func (candidate *CurrentRoundState) MatchesBlock(blockHeader types.Header, round int32) bool {
```

Comment:
is it possible to have different `ResultsHash` with the same `AppHash` and vice versa?